### DISCUSSION TITLE
当前Fake DNS的配置方式非常不合理（附建议）

### DISCUSSION THREAD

**Discussion by bhoppi** [XTLS/Xray-core#391] [Ideas]
我在V2Ray那边发了个讨论帖，考虑到Xray也有这个问题，在这儿引个流，希望各位也来讨论一下。
原帖地址：
https://github.com/v2fly/v2ray-core/discussions/789

**Comment by ghost** [comment]
我认为可以重构 路由， dns 两个模块，并完善 sniffer 与 fakedns。但是工作量较大，优先级不会很高。
